code/datums/elements/poster_tearer.dm

```diff
@@ -0,0 +1,45 @@
+/// Allows mobs with this element attached to just simply tear down any poster they desire to.
+/datum/element/poster_tearer
+	element_flags = ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+	/// The amount of time it takes to tear down a poster.
+	var/tear_time
+	/// Interaction key to use whilst tearing down a poster.
+	var/interaction_key
+
+/datum/element/poster_tearer/Attach(datum/target, tear_time = 2 SECONDS, interaction_key = null)
+	. = ..()
+	if (!isliving(target))
+		return ELEMENT_INCOMPATIBLE
+
+	src.tear_time = tear_time
+	src.interaction_key = interaction_key
+
+	RegisterSignals(target, list(COMSIG_HOSTILE_PRE_ATTACKINGTARGET, COMSIG_LIVING_UNARMED_ATTACK), PROC_REF(on_attacked_poster))
+
+/datum/element/poster_tearer/Detach(datum/source)
+	. = ..()
+	UnregisterSignal(source, list(COMSIG_HOSTILE_PRE_ATTACKINGTARGET, COMSIG_LIVING_UNARMED_ATTACK))
+
+/// Try to tear up a poster on the wall
+/datum/element/poster_tearer/proc/on_attacked_poster(mob/living/user, atom/target, proximity_flag)
+	SIGNAL_HANDLER
+	if(!istype(target, /obj/structure/sign/poster))
+		return NONE // don't care we move on
+
+	if(DOING_INTERACTION_WITH_TARGET(user, target) || (!isnull(interaction_key) && DOING_INTERACTION(user, interaction_key)))
+		user.balloon_alert(target, "busy!")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	INVOKE_ASYNC(src, PROC_REF(tear_it_down), user, target)
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/// Actually work on tearing down that poster
+/datum/element/poster_tearer/proc/tear_it_down(mob/living/user, obj/structure/sign/poster/target)
+	if(!target.check_tearability(user)) // this proc will handle user feedback
+		return
+
+	target.balloon_alert(user, "tearing down the poster...")
+	if(!do_after(user, tear_time, target, interaction_key = interaction_key)) // just in case the user actually enjoys art
+		return
+	target.tear_poster(user)
```

Comment:
This is basically all like "the poster's business" right? 
Why not make this a trait we add to the mob, and then check for it in posters themselves? is there somethin unique that you want to do by keeping it pulled apart like this?

Reply:
I was pondering about doing that and had another branch where I implemented that but I think that this format has the advantage of being able to set the amount of time a mob needs to tear down a poster (just to keep some kayfabe if people want smaller animals to tear down posters) and the interaction keys.

I'm also not interested in implementing three different proc-overrides on poster code to listen for the basic mob/otherwise attack stuff on posters, it is far more simple to both implement and expand it in the element format as it just requires another entry in the list

Reply:
alllright